clang/test/dpct/adl/global_test.cu

```diff
@@ -0,0 +1,116 @@
+// RUN: dpct --format-range=none -out-root %T/adl %s --cuda-include-path="%cuda-path/include"
+// RUN: FileCheck %s --match-full-lines --input-file %T/adl/global_test.dp.cpp
+
+#include <cuda.h>
+#include <iostream>
+
+namespace test {
+template <class T>
+void norm2(T *p) {
+}
+} // namespace test
+namespace hello {
+// CHECK: void norm2(sycl::float3 test) {
+__device__ void norm2(float3 test) {
+}
+} // namespace hello
+
+// CHECK: void norm2(sycl::float3 test) {
+__device__ void norm2(float3 test) {
+}
+
+__device__ void norm2(int a, float3 test) {
+}
+
+__device__ void norm2(float3 test, int a) {
+}
+
+template <class T>
+__device__ void test_norm2(T test) {
+}
+
+__global__ void global_test() {
+  float3 f3;
+  // CHECK: norm2(f3);
+  norm2(f3);
+}
+
+namespace ns {
+__global__ void func();
+};
+__global__ void ns::func() {
+  float3 f3;
+  // CHECK: ::norm2(f3);
```

Comment:
Please add a comment in the test, to make the test easy to understand on the difference introduced by extra "::".